lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -2551,3 +2567,72 @@ instance FromJSON (ApiT SmashServer) where
     parseJSON = fromTextJSON "SmashServer"
 instance ToJSON (ApiT SmashServer) where
     toJSON = toTextJSON
+
+{-------------------------------------------------------------------------------
+                         Token forging types
+-------------------------------------------------------------------------------}
+
+data ForgeTokenData (n :: NetworkDiscriminant) = ForgeTokenData
+    { forgePayments :: !(NonEmpty (AddressForgeAmount (ApiT Address, Proxy n)))
+    , assetName :: !(ApiT W.TokenName)
+    , passphrase :: !(ApiT (Passphrase "lenient"))
+    , metadata :: !(Maybe (ApiT TxMetadata))
+    , timeToLive :: !(Maybe (Quantity "second" NominalDiffTime))
+    } deriving (Eq, Generic, Show)
+
+data AddressForgeAmount addr = AddressForgeAmount
+  { address :: !addr
+  , amt :: !(Quantity "token-unit" Natural)
```

Comment:
Just curious: could the amount type be replaced with `ApiT TokenQuantity` here? (I'm guessing that this is basically the same thing, but correct me if I'm wrong!)

Also, since this API is destined for public consumption, perhaps we should use `amount` rather than an abbreviated form. What do you think?